src/lp_info.rs

```diff
@@ -0,0 +1,628 @@
+    .unwrap()
+});
+
+/// Find spotify playlist URI and extract the album ID
+fn match_spotify_playlist(string: &str) -> Option<&str> {
+    SPOTIFY_PLAYLIST_RE
+        .captures(string.as_ref())
+        .map(|caps| caps.get(1).unwrap().as_str())
+}
+
+#[derive(Command, Debug)]
+#[cmd(name = "lp_info", desc = "Check if listening party is going")]
+pub struct CurrentLP {
+    #[cmd(desc = "Should the answer be visible to everyone?")]
+    visible: Option<bool>,
+}
+
+#[async_trait]
+impl BotCommand for CurrentLP {
+    type Data = Handler;
+    async fn run(
+        self,
+        data: &Handler,
+        ctx: &Context,
+        interaction: &CommandInteraction,
+    ) -> anyhow::Result<CommandResponse> {
+        let mut msg = {
+            // Find last LP
+            let lps = data
+                .module::<ModLPInfo>()
+                .unwrap()
+                .last_pinged
+                .read()
+                .unwrap();
+            let lp = lps.get(&interaction.channel_id);
+            match lp {
+                None => CreateInteractionResponseMessage::new()
+                    .content("There is no listening party at the moment."),
+                Some(lpinfo) => CreateInteractionResponseMessage::new()
+                    .add_embed(lpinfo.build_info_embed()),
+            }
+        };
+
+        if !self.visible.unwrap_or(false) {
+            msg = msg.flags(InteractionResponseFlags::EPHEMERAL);
+        }
+        interaction
+            .create_response(&ctx.http, CreateInteractionResponse::Message(msg))
+            .await?;
+        Ok(CommandResponse::None)
+    }
+}
+
+#[derive(Command, Debug)]
+#[cmd(name = "lp_join", desc = "Join a listening party (privately)")]
+pub struct JoinLP {
+    #[cmd(desc = "Seconds to start playing")]
+    offset: Option<u64>,
+}
+
+#[async_trait]
+impl BotCommand for JoinLP {
+    type Data = Handler;
+    async fn run(
+        self,
+        data: &Handler,
+        ctx: &Context,
+        interaction: &CommandInteraction,
+    ) -> anyhow::Result<CommandResponse> {
+        let offset =
+            chrono::Duration::seconds(self.offset.unwrap_or(15) as i64);
+        let msg = {
+            // Find last LP
+            let lps = data
+                .module::<ModLPInfo>()
+                .unwrap()
+                .last_pinged
+                .read()
+                .unwrap();
+            let lp = lps.get(&interaction.channel_id);
+            match lp {
+                None => CreateInteractionResponseMessage::new()
+                    .content("There is no listening party at the moment."),
+                Some(lpinfo) => CreateInteractionResponseMessage::new()
+                    .add_embed(lpinfo.build_join_embed(offset)),
+            }
+        }
+        .flags(InteractionResponseFlags::EPHEMERAL);
+
+        interaction
+            .create_response(&ctx.http, CreateInteractionResponse::Message(msg))
+            .await?;
+        Ok(CommandResponse::None)
```

Comment:
I pushed changes to make this easier to express.
You should now be able to do:

```suggestion
        // Find last LP
        let lps = data
            .module::<ModLPInfo>()
            .unwrap()
            .last_pinged
            .read()
            .unwrap();
        let lp = lps.get(&interaction.channel_id);
        match lp {
            None => CommandResponse::private("There is no listening party at the moment."),
            Some(lpinfo) => CommandResponse::private(lpinfo.build_join_embed(offset)),
        }
```

Reply:
Ah, cool, I was thinking of writing a PR for this, but part of the reason I didn't was because I was thinking of adding buttons ("join" on the info and "refresh" on the join embed) which would mean I needed to go the manual path anyway. However, implementing component interactions ended up complex and I'd rather get a working version done first. So I'll be using this, thanks!

Reply:
Done